crypto/keys/ed25519/ed25519_test.go

```diff
@@ -228,3 +231,80 @@ func TestMarshalAmino_BackwardsCompatibility(t *testing.T) {
 		})
 	}
 }
+
+// TODO - maybe we should move the tests below to `codec_test` package, WDYT?
+func TestMarshalProto(t *testing.T) {
+	require := require.New(t)
+	ccfg := simapp.MakeTestEncodingConfig()
+	privKey := ed25519.GenPrivKey()
+	pk := privKey.PubKey()
+
+	// **** test JSON serialization ****
+
+	pkAny, err := codectypes.NewAnyWithValue(pk)
+	require.NoError(err)
+	bz, err := ccfg.Marshaler.MarshalJSON(pkAny)
+	require.NoError(err)
+
+	var pkAny2 codectypes.Any
+	err = ccfg.Marshaler.UnmarshalJSON(bz, &pkAny2)
+	require.NoError(err)
+	// we before getting a cached value we need to unpack it.
+	// Normally this happens in in types which implement UnpackInterfaces
+	var pkI cryptotypes.PubKey
+	err = ccfg.InterfaceRegistry.UnpackAny(&pkAny2, &pkI)
+	require.NoError(err)
+	var pk2 = pkAny2.GetCachedValue().(cryptotypes.PubKey)
+	require.True(pk2.Equals(pk))
+
+	// **** test binary serialization ****
+
+	bz, err = ccfg.Marshaler.MarshalBinaryBare(pkAny)
+	fmt.Println(bz)
+	require.NoError(err)
+
+	var pkAny3 codectypes.Any
+	err = ccfg.Marshaler.UnmarshalBinaryBare(bz, &pkAny3)
+	require.NoError(err)
+	err = ccfg.InterfaceRegistry.UnpackAny(&pkAny3, &pkI)
+	require.NoError(err)
+	var pk3 = pkAny3.GetCachedValue().(cryptotypes.PubKey)
+	require.True(pk3.Equals(pk))
+}
+
+func TestMarshalProto2(t *testing.T) {
```

Comment:
It's very hard to understand the difference between the 2 tests, can you think of a better name?

Reply:
I will use TestMarshalProto and TestMarshalProtoInterface.
Feel free to share suggestions